### ISSUE TITLE
Upgrade to Node.js 16

### ISSUE THREAD

**Issue by srich-kr** [MartijnHols/actions-cache#5]
I'm getting this warning in all of my pipelines. 
`Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: martijnhols/actions-cache`

https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

Posting it here for awareness.

**Comment by MartijnHols** [comment]
I just merged all changes up to 3.0.11 of actions/cache into this repository, so this should now be solved.

**Comment by srich-kr** [comment]
That's awesome! Thanks for the quick turnaround! @MartijnHols

**Comment by srich-kr** [comment]
@MartijnHols - That somehow didn't fix the issue yet. 
I was going through the repo and I saw this. 
https://github.com/MartijnHols/actions-cache/blob/main/check/action.yml#L17
https://github.com/MartijnHols/actions-cache/blob/main/restore/action.yml#L23
https://github.com/MartijnHols/actions-cache/blob/main/save/action.yml#L15

We'll have to update these too, probably.

**Comment by MartijnHols** [comment]
Woops, I never considered those files. Thanks for pointing me in the right direction. I only verified if the tests succeeded and one of my complex workflows worked on the latest version before deploying. I never thought to check the annotations as well (though I must admit I haven't updated any other steps to node16 yet). Sorry about that. I fixed it for real now and this time I made sure to verify the annotations disappeared.
